scripts/uploadAgreements.js

```diff
@@ -0,0 +1,85 @@
+/* eslint-disable no-console */
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+const ipfsAPI = require("ipfs-api");
+const fs = require("fs");
+const path = require("path");
+
+const isFilePinned = async (ipfs, hash) => {
+  const matchedHashes = (await ipfs.pin.ls()).filter(response => {
+    if (response.hash === hash) return true;
+    return false;
+  });
+  return matchedHashes.length > 0;
+};
+
+const addFiletoIpfs = async (ipfs, file, name) => {
+  try {
+    let fileHash = (await ipfs.files.add(file, { "only-hash": true }))[0].hash;
+
+    if (await isFilePinned(ipfs, await fileHash)) {
+      await console.log(`file:${name} -- hash:${fileHash} already on ipfs `);
+      return false;
+    }
+    console.log("Adding new file to IPFS and pinning");
+    fileHash = (await ipfs.files.add(file, { pin: true }))[0].hash;
+    console.log(`checking if file was pinned..`);
+    if (!await isFilePinned(ipfs, fileHash))
+      throw new Error("File not succsseffully pinned");
+    console.log("Done");
+    return fileHash;
+  } catch (err) {
+    console.log(err);
+  }
+  return false;
+};
+
+const loadFiles = filePaths =>
+  filePaths.map(filePath => {
+    try {
+      return {
+        name: path.parse(filePath).base,
+        file: fs.readFileSync(filePath)
+      };
+    } catch (e) {
+      throw new Error(`Can't read file "${filePath}"`);
+    }
+  });
+
+// TODO: handle diffrent ports for ipfs
+const main = async ([ipfsNodeAddress, ...paths]) => {
+  const defaultDirpath = path.join(__dirname, "..", "legal");
+  const filePaths = [];
+  const defaultfilePaths = [
```

Comment:
`defaultFileNames` paths are not names. also mind capitalization

Reply:
Done